servers/rendering/renderer_rd/renderer_scene_render_rd.cpp

```diff
@@ -3419,9 +3424,19 @@ void RendererSceneRenderRD::_setup_lights(const PagedArray<RID> &p_lights, const
 						Transform3D modelview = (inverse_transform * li->shadow_transform[j].transform).inverse();
 
 						CameraMatrix shadow_mtx = rectm * bias * matrix * modelview;
+
 						light_data.shadow_split_offsets[j] = split;
-						float bias_scale = li->shadow_transform[j].bias_scale;
-						light_data.shadow_bias[j] = storage->light_get_param(base, RS::LIGHT_PARAM_SHADOW_BIAS) / 100.0 * bias_scale;
+
+						float bias_scale;
+						if (directional_shadow.size < 4096) {
+							// For shadow map sizes below the default, don't increase bias too much to avoid peter-panning.
```

Comment:
Technically, we should be using the same formula as with shadow sizes above the default. However, this would result in too much peter-panning, so I went for a half-way compromise between not adjusting the bias at all and adjusting it to match the resolution perfectly.

Reply:
This should be different for Point Light Shadows, as you have to have the resolution of the cell the shadow is using, not the entire shadow map.

This would incorrectly change bias for the same effective resolution of shadow if you go from a 4k shadow map with 4 4 4 4 quadrants and change to 8k shadow map resolution with 16 16 16 16 quadrants which effectively results in same resolution, just more shadows available.